src/librustc_metadata/creader.rs

```diff
@@ -624,8 +624,12 @@ impl<'a> CrateLoader<'a> {
         impl Registry for MyRegistrar {
             fn register_custom_derive(&mut self,
                                       trait_name: &str,
-                                      expand: fn(TokenStream) -> TokenStream) {
-                let derive = SyntaxExtension::CustomDerive(Box::new(CustomDerive::new(expand)));
+                                      expand: fn(TokenStream) -> TokenStream,
+                                      attributes: &[&'static str]) {
+                let attrs: Vec<_> = attributes.iter().map(|s| InternedString::new(s)).collect();
```

Comment:
nit: `.map(|s| InternedString::new(s))` -> `.map(InternedString::new)`
Also, it `InternedString` is going to become `Name` soon, so it would be better for `attrs` to be `Vec<Name>` instead of `Vec<InternedString>`. In that case, you would use  `.map(token::intern)`.